File: android/app/src/main/kotlin/in/cashify/androidtrc/module/inventory_manager/api/response/PendingPartListResponse.kt
package `in`.cashify.androidtrc.module.inventory_manager.api.response

import `in`.cashify.androidtrc.common.api.BaseResponse
import `in`.cashify.androidtrc.module.inventory_manager.api.PartStatus
import com.google.gson.annotations.SerializedName

class PendingPartListResponse  : BaseResponse() {

    @SerializedName("r_id")
    var rid: String? = null


    @SerializedName("dt")
    var data: ArrayList<Data>? = null


    @SerializedName("s")
   var  status: Boolean? = null


    class Data : BaseResponse() {

        @SerializedName("sku")
        var sku:String? = null

        @SerializedName("pn")
        var pn:String? =null

        @SerializedName("st")
        var st:String? =  null


        @SerializedName("prid")
        var prid:Int? = null


        @SerializedName("stc")
        var statusCode:Int? = null

        @SerializedName("prdt")
        var requestedTime:Long = 0




        fun getPartStatus():PartStatus {
            if(statusCode == 12){
                return PartStatus.AVAILABBLE
            }
            else   if(statusCode == 13){
                return PartStatus.NOT_AVAILABLE
            }


            else{
                return PartStatus.OTHER
            }

        }





    }











}

File: lib/src/modules/inventory_manager/models/pending_part_list_response.dart
import 'package:json_annotation/json_annotation.dart';

import '../resources/part_status_enum.dart';

part 'pending_part_list_response.g.dart';

@JsonSerializable()
class PendingPartListResponse {
  @JsonKey(name: "r_id")
  String? refId;
  @JsonKey(name: "s")
  bool? isSuccess;
  @JsonKey(name: "dt")
  List<PendingPartDataResponse>? partDataList;

  PendingPartListResponse({
    this.isSuccess,
    this.refId,
    this.partDataList,
  });

  static PendingPartListResponse fromJson(Map<String, dynamic> data) => _$PendingPartListResponseFromJson(data);

  Map<String, dynamic> toJson() => _$PendingPartListResponseToJson(this);
}

@JsonSerializable()
class PendingPartDataResponse {
  @JsonKey(name: "sku")
  String? sku;
  @JsonKey(name: "pn")
  String? pn;
  @JsonKey(name: "st")
  String? st;
  @JsonKey(name: "prid")
  int? prid;
  @JsonKey(name: "stc")
  int? statusCode;

  @JsonKey(name: "prdt")
  int? requestedTime;

  PendingPartDataResponse({
    this.sku,
    this.pn,
    this.prid,
    this.requestedTime,
    this.st,
    this.statusCode,
  });

  PartStatus getPartStatus() {
    if (statusCode == 12) {
      return PartStatus.AVAILABLE;
    } else if (statusCode == 13) {
      return PartStatus.NOT_AVAILABLE;
    } else {
      return PartStatus.OTHER;
    }
  }

  static PendingPartDataResponse fromJson(Map<String, dynamic> data) => _$PendingPartDataResponseFromJson(data);

  Map<String, dynamic> toJson() => _$PendingPartDataResponseToJson(this);
}
